pkg/services/ngalert/api/authorization.go

```diff
@@ -162,3 +171,90 @@ func (api *API) authorize(method, path string) web.Handler {
+	namespaceScope := dashboards.ScopeFoldersProvider.GetResourceScope(strconv.FormatInt(namespace.Id, 10))
+	if len(changes.Delete) > 0 {
+		allowed, err := evaluator(ac.EvalPermission(ac.ActionAlertingRuleDelete, namespaceScope))
+		if err != nil {
+			return fmt.Errorf("failed to perform user authorization: %w", err)
+		}
+		if !allowed {
+			return fmt.Errorf("%w user cannot delete alert rules that belong to folder %s", ErrAuthorization, namespace.Title)
+		}
+	}
+
+	// check that the user can create alerts in
+	if len(changes.New) > 0 {
+		allowed, err := evaluator(ac.EvalPermission(ac.ActionAlertingRuleCreate, namespaceScope))
+		if err != nil {
+			return fmt.Errorf("failed to perform user authorization: %w", err)
+		}
+		if !allowed {
+			return fmt.Errorf("%w user cannot create alert rules in the folder %s", ErrAuthorization, namespace.Title)
+		}
+	}
+
+	updateChecked := false
+
+	for _, rule := range changes.Update {
+		if rule.Existing.NamespaceUID != rule.New.NamespaceUID {
+			allowed, err := evaluator(ac.EvalAll(ac.EvalPermission(ac.ActionAlertingRuleDelete, dashboards.ScopeFoldersProvider.GetResourceScopeUID(rule.Existing.NamespaceUID))))
+			if err != nil {
+				return fmt.Errorf("failed to perform user authorization: %w", err)
+			}
+			if !allowed {
+				return fmt.Errorf("%w to delete alert rules from folder UID %s", ErrAuthorization, rule.Existing.NamespaceUID)
+			}
+			allowed, err = evaluator(ac.EvalPermission(ac.ActionAlertingRuleCreate, namespaceScope))
+			if err != nil {
+				return fmt.Errorf("failed to perform user authorization: %w", err)
+			}
+			if !allowed {
+				return fmt.Errorf("%w to create alert rules in the folder '%s'", ErrAuthorization, namespace.Title)
+			}
+		} else if !updateChecked {
+			allowed, err := evaluator(ac.EvalAll(ac.EvalPermission(ac.ActionAlertingRuleUpdate, namespaceScope)))
+			if err != nil {
+				return fmt.Errorf("failed to perform user authorization: %w", err)
+			}
+			if !allowed {
+				return fmt.Errorf("%w to update alert rules that belong to folder %s", ErrAuthorization, namespace.Title)
+			}
+			updateChecked = true
+		}
+
+		// if data were not changed, skip authorization. Effectively, this will let users without read access for
+		// data sources limited set of settings, like interval, for, labels, annotations.
+		if len(rule.Diff.GetDiffsForField("Data")) > 0 {
+			dsAllowed, err := evaluator(getEvaluatorForAlertRule(rule.New))
+			if err != nil {
+				return fmt.Errorf("failed to perform user authorization: %w", err)
+			}
+			if !dsAllowed {
+				action := "create"
+				title := rule.New.Title
+				if rule.Existing != nil {
+					action = "update"
+					title = fmt.Sprintf("%s (UID: %s)", rule.Existing.Title, rule.Existing.UID)
+				}
+				return fmt.Errorf("%w to %s alert rule %s because the does not have read permissions for one or many datasources the rule uses", ErrAuthorization, action, title)
```

Comment:
```suggestion
				return fmt.Errorf("%w to %s alert rule %s because the user does not have read permissions for one or many datasources the rule uses", ErrAuthorization, action, title)
```